server/channels/app/platform/web_conn.go

```diff
@@ -253,7 +253,10 @@ func (ps *PlatformService) NewWebConn(cfg *WebConnConfig, suite SuiteIFace, runn
 		// Create a goroutine to avoid blocking the creation of the websocket connection.
 		ps.Go(func() {
 			ps.SetStatusOnline(userID, false)
-			ps.UpdateLastActivityAtIfNeeded(*wc.GetSession())
+			session := wc.GetSession()
+			if session != nil {
+				ps.UpdateLastActivityAtIfNeeded(*session)
+			}
```

Comment:
Would simply using the original `cfg.Session` cause more problems here?

Reply:
No, not using the session from `cfg. Session` was the whole point of https://github.com/mattermost/mattermost/pull/28195. At the point when `UpdateLastActivityAtIfNeeded` is executed, the session from `cfg. Session` might already be part of the `sync. Pool`, and another go routine could have ownership of it.

Reply:
I see, we are indirectly using the deep copy through `wc.GetSession()`. 

@agnivade Do you feel there's still value in having a pool if we end up deep copying on every new connection?

Reply:
We are deep copying only for websocket requests, not for plain HTTP requests. So think a pool still makes sense.